x-pack/filebeat/input/netflow/integration_test.go

```diff
@@ -0,0 +1,388 @@
+		}
+	}
+
+	return -1, nil
+}
+
+type DataStream struct {
+	Name   string `json:"name"`
+	Status string `json:"status"`
+}
+
+type DataStreamResult struct {
+	DataStreams []DataStream `json:"data_streams"`
+	Error       interface{}  `json:"error"`
+}
+
+func HasDataStream(username string, password string, url string, name string) error {
+	resultBytes, err := request(http.MethodGet, username, password, fmt.Sprintf("%s/_data_stream/%s", url, name))
+	if err != nil {
+		return err
+	}
+
+	if resultBytes == nil {
+		return errors.New("http not found error")
+	}
+
+	var results DataStreamResult
+	err = json.Unmarshal(resultBytes, &results)
+	if err != nil {
+		return err
+	}
+
+	if results.Error != nil {
+		return fmt.Errorf("error %v while checking for data stream %s", results.Error, name)
+	}
+
+	if len(results.DataStreams) != 1 {
+		return fmt.Errorf(
+			"unexpected count %v of data streams returned when looking for %s",
+			len(results.DataStreams), name)
+	}
+
+	if results.DataStreams[0].Name != name {
+		return fmt.Errorf("unexpected data stream %s returned when looking for %s",
+			results.DataStreams[0].Name,
+			name)
+	}
+
+	return nil
+}
+
+// Hit represents a single search hit.
+type Hit struct {
+	Index  string                 `json:"_index"`
+	Type   string                 `json:"_type"`
+	ID     string                 `json:"_id"`
+	Score  float64                `json:"_score"`
+	Source map[string]interface{} `json:"_source"`
+}
+
+// Hits are the collections of search hits.
+type Hits struct {
+	Total json.RawMessage // model when needed
+	Hits  []Hit           `json:"hits"`
+}
+
+// SearchResults are the results returned from a _search.
+type SearchResults struct {
+	Took   int
+	Hits   Hits                       `json:"hits"`
+	Shards json.RawMessage            // model when needed
+	Aggs   map[string]json.RawMessage // model when needed
+}
+
+func DataStreamEventsCount(username string, password string, url string, name string) (int, error) {
+	resultBytes, err := request(http.MethodGet, username, password, fmt.Sprintf("%s/%s/_search?q=!error.message:*", url, name))
+	if err != nil {
+		return 0, err
+	}
+
+	if resultBytes == nil {
+		return 0, errors.New("http not found error")
+	}
+
+	var results SearchResults
+	err = json.Unmarshal(resultBytes, &results)
+	if err != nil {
+		return 0, err
+	}
+	return len(results.Hits.Hits), nil
+}
+
+// DeleteResults are the results returned from a _data_stream delete.
+type DeleteResults struct {
+	Acknowledged bool `json:"acknowledged"`
+}
+
+func DeleteDataStream(username string, password string, url string, name string) (bool, error) {
+	_, err := request(http.MethodDelete, username, password, fmt.Sprintf("%s/_data_stream/%s", url, name))
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func request(httpMethod string, username string, password string, url string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(context.TODO(), httpMethod, url, nil)
```

Comment:
I was using `http.NewRequest()` but the linter was failing because I wasn't using NewRequestWithContext. So I was kinda forced to use a context.TODO(). Now it terms of meaningfulness `context.TODO` I think describes really well my intentions https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/context/context.go;l=219